src/Disks/IO/AsynchronousReadIndirectBufferFromRemoteFS.cpp

```diff
@@ -146,46 +146,45 @@ bool AsynchronousReadIndirectBufferFromRemoteFS::nextImpl()
         return false;
 
     size_t size = 0;
-
     if (prefetch_future.valid())
     {
         ProfileEvents::increment(ProfileEvents::RemoteFSPrefetchedReads);
 
-        CurrentMetrics::Increment metric_increment{CurrentMetrics::AsynchronousReadWait};
-        Stopwatch watch;
+        size_t offset = 0;
         {
+            Stopwatch watch;
+            CurrentMetrics::Increment metric_increment{CurrentMetrics::AsynchronousReadWait};
             auto result = prefetch_future.get();
             size = result.size;
-            auto offset = result.offset;
-            assert(offset < size);
-
-            if (size)
-            {
-                memory.swap(prefetch_buffer);
-                size -= offset;
-                set(memory.data() + offset, size);
-                working_buffer.resize(size);
-                file_offset_of_buffer_end += size;
-            }
+            offset = result.offset;
+            assert(offset < size || size == 0);
```

Comment:
I would rather remove `|| size == 0` here. It is not needed (because of checks for remainining size to read here we can expect exactly `offset < size && size != 0`) and it helps to find the source of some bugs much quicker.

Reply:
I see. However we don't have such invariant in `AsynchronousReadBufferFromFileDescriptor`. I guess that's different.